trieste/bayesian_optimizer.py

```diff
@@ -182,7 +183,20 @@ def optimize(
         num_steps: int,
         datasets: Mapping[str, Dataset],
         model_specs: Mapping[str, ModelSpec],
-        acquisition_rule: AcquisitionRule[S, SP],
+        acquisition_rule: AcquisitionRule[TensorType, SP],
```

Comment:
unfortunately this clashes with the other overload. I believe this is because `TensorType` is untyped. I'm not sure what to do about that